wrapanapi/nuage.py

```diff
@@ -53,26 +65,206 @@ def common_headers(self):
+        return self._request_list('/domains/{}/subnets'.format(domain), 'get') or []
+
+    def list_cloud_subnets_for_zone(self, zone):
+        return self._request_list('/zones/{}/subnets'.format(zone), 'get') or []
 
     def list_security_groups(self):
         return self._request_list('/policygroups', 'get')
 
+    def list_enterprises(self, name=None):
+        return self._request_list('/enterprises', 'get', search_name=name) or []
+
+    def list_domain_templates(self, enterprise=None, name=None):
+        if not enterprise:
+            enterprise = self.enterprise
+        return self._request_list('/enterprises/{}/domaintemplates'.format(enterprise), 'get',
+                             search_name=name) or []
+
+    def list_domains(self, name=None):
+        return self._request_list('/domains', 'get', search_name=name) or []
+
+    def list_domains_for_enterprise(self, enterprise=None, name=None):
+        if not enterprise:
+            enterprise = self.enterprise
+        return self._request_list(
+            '/enterprises/{}/domains'.format(enterprise),
+            'get',
+            search_name=name
+        ) or []
+
+    def list_zones_for_domain(self, domain=None, name=None):
+        if not domain:
+            domain = self.domain
+        return self._request_list('/domains/{}/zones'.format(domain), 'get', search_name=name) or []
+
+    def create_enterprise(self):
+        return self._request('/enterprises', 'post', data={
+            'name': self.enterprise_name,
+            'description': self.entity_description
+        })
+
+    def create_domain_template(self, enterprise=None):
+        if not enterprise:
+            enterprise = self.enterprise
+        return self._request('/enterprises/{}/domaintemplates'.format(enterprise), 'post', data={
+            'name': self.domain_template_name,
+            'description': self.entity_description
+        })
+
+    def create_domain(self, enterprise=None, domain_template=None):
+        if not enterprise:
+            enterprise = self.enterprise
+        if not domain_template:
+            domain_template = self.domain_template
+        return self._request('/enterprises/{}/domains'.format(enterprise), 'post', data={
+            'name': self.domain_name,
+            'description': self.entity_description,
+            'templateID': domain_template
+        })
+
+    def create_zone(self, domain=None):
+        if not domain:
+            domain = self.domain
+        return self._request('/domains/{}/zones'.format(domain), 'post', data={
+            'name': self.zone_name,
+            'description': self.entity_description
+        })
+
+    def create_subnet(self, zone=None):
+        if not zone:
+            zone = self.zone
+        return self._request('/zones/{}/subnets'.format(zone), 'post', data={
+            'name': self.subnet_name,
+            'description': self.entity_description,
+            'address': '193.192.191.0',
```

Comment:
This information should not be hardcoded, and should be passed to create_subnet as kwargs